libs/blockchain/src/blockchain.service.ts

```diff
@@ -178,30 +183,44 @@ export class BlockchainService extends BlockchainRpcQueryService implements OnAp
       throw new DelayedError();
     }
     const keys = new Keyring({ type: 'sr25519' }).createFromUri(this.config.providerSeedPhrase);
-    const nonce = await this.getNextNonce();
-    this.logger.debug(`Capacity Wrapped Extrinsic: ${JSON.stringify(extrinsic.toHuman())}, nonce: ${nonce}`);
-    const txHash = await extrinsic.signAndSend(keys, { nonce });
-    if (!txHash) {
-      throw new Error('Tx hash is undefined');
+    const nonce = await this.reserveNextNonce();
+    try {
+      this.logger.debug(`Capacity Wrapped Extrinsic: ${JSON.stringify(extrinsic.toHuman())}, nonce: ${nonce}`);
+      const txHash = await extrinsic.signAndSend(keys, { nonce });
+      if (!txHash) {
+        throw new Error('Tx hash is undefined');
+      }
+      this.logger.debug(`Tx hash: ${txHash.toString()}`);
+      return [extrinsic, txHash.toHex()];
+    } catch (err: any) {
+      this.unreserveNonce(nonce);
```

Comment:
This might be intentional but It might make sense to wait for it. It might prevent some funky edge cases
```suggestion
      await this.unreserveNonce(nonce);
```